src/main/java/org/broadinstitute/hellbender/utils/codecs/sampileup/SAMPileupFeature.java

```diff
@@ -0,0 +1,136 @@
+package org.broadinstitute.hellbender.utils.codecs.sampileup;
+
+import htsjdk.samtools.SAMUtils;
+import htsjdk.samtools.util.StringUtil;
+import htsjdk.tribble.Feature;
+import org.apache.commons.lang.ArrayUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A tribble feature representing a SAM pileup.
+ *
+ * Allows intake of simple mpileups. Simple pileup features will contain only basic information, no reconstructed reads.
+ *
+ * @author Danil Gomez-Sanchez (magiDGS)
+ */
+public class SAMPileupFeature implements Feature {
+
+    // genomic location
+    private final String contig;
+    private final int position;
+    // reference base
+    private final byte refBase;
+
+    // list of pileup elements
+    private final List<SAMPileupElement> pileupElements;
+
+    SAMPileupFeature(final String contig, final int position, final byte refBase, final List<SAMPileupElement> pileupElements) {
+        Utils.nonNull(pileupElements);
+        this.contig = contig;
+        this.position = position;
+        this.refBase = refBase;
+        this.pileupElements = pileupElements;
+    }
+
+    @Override
+    @Deprecated
+    public String getChr() {
+        return getContig();
+    }
+
+    @Override
+    public String getContig() {
+        return contig;
+    }
+
+    @Override
+    public int getStart() {
+        return position;
+    }
+
+    @Override
+    public int getEnd() {
+        return position;
+    }
+
+    /**
+     * Returns pile of obseved qualities over the genomic location
+     *
+     * Note: this call costs O(n) and allocates fresh array each time
+     */
+    public String getQualsAsString() {
+        return SAMUtils.phredToFastq(getBases());
+    }
+
+    /**
+     * Returns pile of observed bases over the genomic location.
+     *
+     * Note: this call costs O(n) and allocates fresh array each time
+     */
+    public String getBasesAsString() {
+        return StringUtil.bytesToString(getBases());
+    }
+
+    /**
+     * Returns the reference basse
+     */
+    public byte getRef() {
+        return refBase;
+    }
+
+    /**
+     * Return the number of observed bases over the genomic location
+     */
+    public int size() {
+        return pileupElements.size();
+    }
+
+    /**
+     * Returns formatted pileup string for the genomic location as
+     * "location: reference_base observed_base_pile observed_qual_pile"
+     */
+    public String getPileupString() {
```

Comment:
It is weird that `ReadPileup` and `SAMPileupFeature`, which are similar, have a different behavior regarding this method. I changed now the contract and added a test for it (and solve a bug that I detected in getQualsAsString).